src/main/scala/com/databricks/labs/mosaic/functions/MosaicContext.scala

```diff
@@ -497,6 +502,7 @@ class MosaicContext(indexSystem: IndexSystem, geometryAPI: GeometryAPI) extends
         def st_centroid2D(geom: Column): Column = ColumnAdapter(ST_Centroid(geom.expr, geometryAPI.name))
         def st_centroid3D(geom: Column): Column = ColumnAdapter(ST_Centroid(geom.expr, geometryAPI.name, 3))
         def st_convexhull(geom: Column): Column = ColumnAdapter(ST_ConvexHull(geom.expr, geometryAPI.name))
+        def st_difference(geom1: Column, geom2: Column) = ColumnAdapter(ST_Difference(geom1.expr, geom2.expr, geometryAPI.name))
```

Comment:
Can this line be updated to have the return type specified? this is what is breaking the R build

Reply:
`def st_difference(geom1: Column, geom2: Column): Column`

Reply:
in fact, it would be helpful to put a comment at the top of the functions object definition highlighting the expected pattern, i.e. 

// function definitions should follow the below pattern
// def name(argName: Type, ...): returnType = ...

Reply:
I've updated the specific line and added the suggested top level comment.